runtime/src/main/java/org/corfudb/runtime/collections/CorfuStore.java

```diff
@@ -261,9 +261,9 @@ public long getHighestSequence(@Nonnull final String namespace,
         corfuStoreMetrics.recordBatchReads(numBatches);
         corfuStoreMetrics.recordNumberReads(numBatches * runtime.getParameters().getHighestSequenceNumberBatchSize());
         corfuStoreMetrics.recordHighestSequenceNumberDuration(startTime);
-        log.debug("Stream[{}${}][{}] no DATA entry found. Highest sequence number corresponds to trim mark={}",
-                namespace, tableName, Utils.toReadableId(streamId), streamAddressSpace.getTrimMark());
-        return streamAddressSpace.getTrimMark();
+        log.debug("Stream[{}${}][{}] no DATA entry found. Returning -1.",
+                namespace, tableName, Utils.toReadableId(streamId));
+        return -1;
```

Comment:
How do you make sure this won't regress in future patches? Should we just assume that it wont, or should we have a unit test and enforce it in code?

Reply:
My bad, added a test and removed it temporarily to triage flaky tests and forgot to add it back..
going to push it in a separate patch.

Reply:
here is the test case https://github.com/CorfuDB/CorfuDB/pull/2966/files#diff-14edd36945df0ba1730f3db2fbfac31222a9d10e289513942c1b1f63f7cdd768R1111